app/views/contexts/summary/answer.py

```diff
@@ -64,17 +64,24 @@ def _build_link(
         return_location: ReturnLocation,
         is_in_repeating_section: bool,
     ) -> str:
+        return_to_answer_id = self._return_to_answer_id(
+            return_to=return_location.return_to,
+            list_item_id=list_item_id,
+            is_in_repeating_section=is_in_repeating_section,
+        )
+        if (
+            return_location.return_to_answer_id
+            and return_to_answer_id
+            not in return_location.return_to_answer_id.split(",")
+        ):
+            return_to_answer_id += f",{return_location.return_to_answer_id}"  # type: ignore
```

Comment:
I think any extra logic should probably be in the `_return_to_answer_id` method

Reply:
Moved it there now.